### ISSUE TITLE
RouterQueryEngine sometimes raises JSONDecodeError during inference

### ISSUE THREAD

**Issue by armen-tractatus** [jerryjliu/llama_index#2808]
We are using a `RouterQueryEngine` with a chat agent initialized with `create_llama_chat_agent`. Sometimes when we query the agent, `agent.run(query)` raises a `JsonDecodeError`. Sometimes the query executes successfully. The code that raises the error is in `selection.py` in the `SelectionOutputParser->parse(output)` method. 

Sometimes `output` is as follows, which fails in `json.loads(output)` because of the extra `Output:` at the start of the `output` string:
```json
Output:
[{
  "choice": 3,
  "reason": "This choice is most relevant to the question as it specifically mentions the <item>."
}]
```

However, sometimes the value of `output` is valid JSON and the query completes successfully:
```json
[{
  "choice": 3,
  "reason": "This choice is most relevant to the question as it is specifically about <product>."
}]
```

The LLM output that causes this issue is in `llm_selectors.py` in `LLMSingleSelector->_select()`:
```python
prediction, _ = self._llm_predictor.predict( ... )
```

This issue can be solved by providing `prompt_template_str` or `output_parser` to the arguments of `LLMSingleSelector->from_defaults()`. We will proceed down that path, but it would have been nice if the defaults were more robust.

cc: @ali-tractatus

**Comment by armen-tractatus** [comment]
I wrote a solution for this that works quite well, but can't push to this repo due to lack of permissions. Let me know if we can help fix this.

**Comment by decent-engineer-decent-datascientist** [comment]
@armen-tractatus try putting in a pull request. What was the solution, if I may ask?

**Comment by armen-tractatus** [comment]
I tried pushing a branch to create a PR, but got a 403 forbidden error. Tried with a token and by logging in to GitHub.

Here was the solution that worked for me. In `llama_index/output_parsers/selection.py`:

Update the `FORMAT_STR` template by clarifying the instructions (do not include prefix)
```python
FORMAT_STR = """The output should be formatted as a JSON instance that conforms to 
the JSON schema below. Do not include the prefix 'Output:' in the answer.
```

That generally fixed the problem (I did this on my side by overriding the `SelectionOutputParser->format()` method and appended that clarification to the very end of the prompt, which also worked,

To make the solution extra robust, I also added this fix in `SelectionOutputParser->parse()`, which removes the `Output:` prefix if it appears at the start of the string:
```python
    def parse(self, output: str) -> Any:
        clean_output = output.replace('Output:', '', 1) if output.strip().startswith('Output:') else output
        json_list = json.loads(clean_output)
```
